instaloader/structures.py

```diff
@@ -373,14 +375,20 @@ def is_video(self) -> bool:
     @property
     def video_url(self) -> Optional[str]:
         """URL of the video, or None."""
+        def get_size(url):
+            return requests.head(url).headers.get('Content-Length', 0), url
         if self.is_video:
+            url_candidates = []
+            if 'video_url' in self._node:
+                url_candidates.append(get_size(self._node['video_url']))
             if self._context.is_logged_in:
                 try:
-                    url = self._iphone_struct['video_versions'][0]['url']
-                    return url
+                    self._iphone_struct['video_versions'].sort(key=lambda x: x['height'])
+                    url_candidates.append(get_size(self._iphone_struct['video_versions'][-1]['url']))
```

Comment:
To me it seems we can just compare them all and choose the one with the biggest filesize, i.e. the highest bitrate:

```suggestion
                    for version in self._iphone_struct['video_versions']:
                        url_candidates.append(get_size(version['url']))
```